views/js/controller/creator/helpers/qtiTest.js

```diff
@@ -217,8 +229,24 @@ define([
          * @throws {Error} if the model is not valid
          */
         validateModel: function validateModel(model) {
+            var identifiers = this.extractIdentifiers(model);
+            var nonUniqueIdentifiers = [];
             var outcomes = _.indexBy(outcomeHelper.listOutcomes(model));
 
+            _(identifiers)
+                .countBy()
+                .forEach(function(count, id){
+                    if(count > 1){
+                        nonUniqueIdentifiers.push(id);
+                    }
+                });
+            if(nonUniqueIdentifiers.length === 1){
+                throw new Error(__('The identifier "%s" is not unique accross the test.', nonUniqueIdentifiers[0]));
+            }
+            if(nonUniqueIdentifiers.length > 1){
+                throw new Error(__('The following identifiers "%s" are not unique accross the test.', nonUniqueIdentifiers.join('", "')));
```

Comment:
Could be more useful, if the error message indicates elements is faulty by highlighting it. But since it is an edge case, could be a separate story, we should as product owner.

Reply:
As a rule of thumb, when displaying an error message, the next step for the user should be clear, obvious and easily doable.

Reply:
Yes I was thinking to it also, but there's no link between the ids and the elements in the current version. 
And I was balancing 2 hours of developments against an error that _should_ never occur, the only way to get there is to have created a test with the bug in the previous version. 
This error message is just a last barrier to not save a broken test.